### PR TITLE
Refactoring and upgrade to latest version - Local branch

### PR DESCRIPTION
**Pull request by maniackcrudelis**
## Problem
- *Package out of date*
- *Doesn't work anymore on stable*

## Solution
- *Major upgrade of Roundcube version*
- *Package refactored to adjust to latest standards*

## PR Status
Work finished  
Package_check OK.  
Basic tests OK.  
Upgrade not tested. The previous version can't be installed.  
Could be reviewed and tested.  

## Validation
~*Medium decision*~
- [ ] ~**Complete test** :~
- [ ] ~**Upgrade previous version** : not testable~
- [X] ~**Code review** : JimboJoe~
- [ ] ~**Code review** :~
- [X] ~**Approval (LGTM)** : JimboJoe~
- [ ] ~**Approval (LGTM)** :~
- [ ] ~**Approval (LGTM)** :~
- [X] ~**CI succeeded** : JimboJoe~
- [ ] ~**CI succeeded** :~
~One public package check log is enough instead of two checks.  
When the PR is mark as ready to merge, you have to wait for 7 days before really merge it.~

Since this PR has been rerouted from master to testing. It passes to a minor decision.
- [x] **Upgrade previous version** : not testable
- [x] **Code review** : JimboJoe
- [x] **Approval (LGTM)** : JimboJoe
- [x] **Approval (LGTM)** : Aleks, Because we have to go forward !
- [x] **CI succeeded** : JimboJoe
When the PR is mark as ready to merge, you have to wait for 3 days before really merge it.

### CODE REVIEW

**Comment by JimboJoe** [comment]
Shouldn't we take advantage of this huge refactoring to adopt latest 2.7.2 standards (no sudo prefixes in every script, remove helpers)?
As official apps are set as examples, I would say a clear YES :wink:

**Comment by maniackcrudelis** [comment]
Yes you're right, I forgot some residual `sudo`
And, for the helpers, just I begin to work on these refactoring before the release. And my VM is not up to date...  
I'm going to fix this on my side.  
Feel free to edit this PR ;)

**Comment by JimboJoe** [comment]
And I also forgot to recommend to implement change_url scripts from now on ;-)
So... you said the PR was ready, but I guess it's not really... :wink:

**Comment by maniackcrudelis** [comment]
I agree, it would be nice to implemente change_url scripts. But, for now, I will just make a refactoring, not any improvement.
Because, we have 20 official apps, and I'm not the maintainer of these apps.
So, I think it's would be great to have, at least, our apps with the last helpers and the correct shape.

**Comment by JimboJoe** [comment]
So... are we ready now for reviewing? :grinning:

**Comment by maniackcrudelis** [comment]
It's ok, we're ready ;)

File: scripts/restore [hunk 1]
```diff
@@ -1,58 +1,95 @@
...
+domain=$(ynh_app_setting_get $app domain)
+path_url=$(ynh_app_setting_get $app path)
+final_path=$(ynh_app_setting_get $app final_path)
+db_name=$(ynh_app_setting_get $app db_name)
+
+#=================================================
+# CHECK IF THE APP CAN BE RESTORED
+#=================================================
+
+ynh_webpath_available $domain $path_url \
```
Comment (JimboJoe) [inline scripts/restore]: Shouldn't we halso register the URL via `ynh_webpath_register`?

Comment (maniackcrudelis) [reply scripts/restore]: Not in the restore script.
It's also for this reason they split the command in 2 differents, because you can't register again the domain.
If my memories are good...

File: check_process [hunk 2]
```diff
@@ -11,10 +11,10 @@
 		setup_nourl=0
 		setup_private=0
 		setup_public=0
-		upgrade=1
+		upgrade=0
```
Comment (maniackcrudelis) [inline check_process]: A little oups ?

File: scripts/restore (hunk 1 above)
Comment (JimboJoe) [reply scripts/restore]: But, you can't restore if the app is already installed... so the URL isn't registered either, is it...? :frowning:

File: check_process (hunk 2 above)
Comment (JimboJoe) [reply check_process]: Indeed!

File: scripts/restore (hunk 1 above)
Comment (maniackcrudelis) [reply scripts/restore]: So, I have to ask to ljf... He's just in front of me.

Comment (maniackcrudelis) [reply scripts/restore]: So, I have some answers !
In fact, `ynh_webpath_available` will just check if the the path is currently available in the sso (or somewhere else ?).
But, not store the information in the settings.yml file.

During the restore process, the settings.yml file is restored at the beginning, but the sso will not be reload until the end of the script. So the sso still consider the path is available.
If you use `ynh_webpath_register`, the sso will be reload immediately, and the path is not longer available.

In conclusion, it's not needed to used `ynh_webpath_register`, because the settings file is already restored. But I think you can if you use it after `ynh_webpath_available`.

Comment (JimboJoe) [reply scripts/restore]: OK, roger that.
So, for what reasons did the checkurl command get split-up?

Comment (maniackcrudelis) [reply scripts/restore]: Because, it's better to check if the path is available. But the old command had the bad habit to fail because the path is already in the settings.yml file.
I don't know much more about this command though...

Comment (maniackcrudelis) [reply scripts/restore]: > There is also bug #983 regarding the use of checkurl in the remove script of example_ynh. If you confirm this PR as an model, we'll have fix the example_ynh bug accordingly.

It's already fixed on my local copy. I made a lot of correction on example_ynh.

**Comment by alexAubin** [comment]
Any update on this ? Is there any way we could test the update from a previous commit in the past or is there no hope at all because the old Composer install is broken anyway now ? (no idea how this all works)

**Comment by alexAubin** [comment]
People still regularly come complain on the support that roundcube is broken... Any chance that we can move forward with this ?

As far as my opinion counts, I'm okay with merging ASAP. I can try an install if that's really needed..

**Comment by maniackcrudelis** [comment]
@alexAubin Let's take your advise as the LGTM we're missing.
And, we can merged it in 3 days in the testing branch

**Comment by alexAubin** [comment]
Agreed ! Cool, thanks :smile:

**Comment by JimboJoe** [comment]
Welcome to @alexAubin in the Apps Group! 🎉 
More seriously, that's a pity we couldn't get a simple LGTM from the Apps Group in 3 weeks... We've got to learn from that for the other apps approval.
